hathor/builder/builder.py

```diff
@@ -299,28 +306,51 @@ def _get_p2p_manager(self) -> ConnectionsManager:
         )
         return p2p_manager
 
-    def _get_or_create_tx_storage(self) -> TransactionStorage:
+    def _get_or_create_indexes_manager(self) -> IndexesManager:
+        if self._indexes_manager is not None:
+            return self._indexes_manager
+
+        if self._force_memory_index or self._storage_type == StorageType.MEMORY:
+            self._indexes_manager = MemoryIndexesManager()
+
+        elif self._storage_type == StorageType.ROCKSDB:
+            rocksdb_storage = self._get_or_create_rocksdb_storage()
+            self._indexes_manager = RocksDBIndexesManager(rocksdb_storage)
+
+        else:
+            raise NotImplementedError
+
+        return self._indexes_manager
+
+    def _get_or_create_tx_storage(self, indexes: IndexesManager) -> TransactionStorage:
         if self._tx_storage is not None:
+            # If a tx storage is provided, set the indexes manager to it.
+            self._tx_storage.indexes = indexes
             return self._tx_storage
 
+        store_indexes: Optional[IndexesManager] = indexes
+        if self._tx_storage_cache:
+            cache_indexes = indexes
+            store_indexes = None
+
         if self._storage_type == StorageType.MEMORY:
-            return TransactionMemoryStorage()
+            self._tx_storage = TransactionMemoryStorage(indexes=store_indexes)
 
-        if self._storage_type == StorageType.ROCKSDB:
+        elif self._storage_type == StorageType.ROCKSDB:
             rocksdb_storage = self._get_or_create_rocksdb_storage()
-            use_memory_index = self._force_memory_index
+            self._tx_storage = TransactionRocksDBStorage(rocksdb_storage, indexes=store_indexes)
 
-            kwargs = {}
-            if self._rocksdb_with_index is not None:
-                kwargs = dict(with_index=self._rocksdb_with_index)
+        else:
+            raise NotImplementedError
 
-            return TransactionRocksDBStorage(
-                rocksdb_storage,
-                use_memory_indexes=use_memory_index,
-                **kwargs
-            )
+        if self._tx_storage_cache:
+            reactor = self._get_reactor()
+            kwargs: Dict[str, Any] = {}
+            if self._tx_storage_cache_capacity is not None:
+                kwargs['capacity'] = self._tx_storage_cache_capacity
+            self._tx_storage = TransactionCacheStorage(self._tx_storage, reactor, indexes=cache_indexes, **kwargs)
```

Comment:
It looks like you don't need the `cache_indexes` variable, you could just use `indexes` and remove it from line 333. Also, you could use `dict` instead of `Dict` for the `kwargs` typing

Reply:
Nice catch! Fixed. Thanks!